src/main/java/graphql/schema/visibility/NoIntrospectionGraphqlFieldVisibility.java

```diff
@@ -12,10 +12,15 @@
  * This field visibility will prevent Introspection queries from being performed.  Technically this puts your
  * system in contravention of <a href="https://spec.graphql.org/October2021/#sec-Introspection">the specification</a>
  * but some production systems want this lock down in place.
+ *
+ * @deprecated This is no longer the best way to prevent Introspection - {@link graphql.introspection.Introspection#enabledJvmWide(boolean)}
+ * can be used instead
  */
 @PublicApi
+@Deprecated // Deprecated since 2024-03-16
 public class NoIntrospectionGraphqlFieldVisibility implements GraphqlFieldVisibility {
 
+    @Deprecated // Deprecated since 2024-03-16
```

Comment:
Java 8 special: the deprecated annotation in Java 8 doesn't have the "since" argument